src/flash/nor/rp2040.c

```diff
@@ -185,6 +185,16 @@ static int rp2040_flash_enter_xip(struct flash_bank *bank)
 	struct rp2040_flash_bank *priv = bank->driver_priv;
 	int err = ERROR_OK;
 
+	// Always flush before returning to execute-in-place, to invalidate stale cache contents.
+	// The flush call also restores regular hardware-controlled chip select following a rp2040_flash_exit_xip().
+	LOG_DEBUG("Flushing flash cache after write behind");
+	err = rp2040_call_rom_func(bank->target, priv->stacktop, FUNC_FLASH_FLUSH_CACHE, NULL, 0);
```

Comment:
Haha, "func flash flush cache" is a bit of a tongue-twister if said out loud :wink: